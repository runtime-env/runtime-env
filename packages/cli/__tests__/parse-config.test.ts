import parseConfig from "../src/parse-config";

const minimalValidConfig = {
  globalVariableName: "_",
  envExampleFilePath: "_",
  genJs: [
    {
      mode: "_",
      envFilePath: null,
      userEnvironment: false,
      outputFilePath: "_",
    },
  ],
};

describe("globalVariableName", () => {
  test("valid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        globalVariableName: "fooBar",
      }),
    ).not.toThrow();
  });

  test("invalid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        globalVariableName: "foo-bar",
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});

describe("genJs.*.mode", () => {
  test("valid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            ...minimalValidConfig.genJs[0],
            mode: "foo",
          },
          {
            ...minimalValidConfig.genJs[0],
            mode: "bar",
          },
        ],
      }),
    ).not.toThrow();
  });

  test("invalid", () => {
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            ...minimalValidConfig.genJs[0],
            mode: "foo",
          },
          {
            ...minimalValidConfig.genJs[0],
            mode: "foo",
          },
        ],
      }),
    ).toThrowErrorMatchingSnapshot();
    expect(() =>
      parseConfig({
        ...minimalValidConfig,
        genJs: [
          {
            ...minimalValidConfig.genJs[0],
            mode: "foo",
          },
          {
            ...minimalValidConfig.genJs[0],
            mode: "bar",
          },
          {
            ...minimalValidConfig.genJs[0],
            mode: "foo",
          },
        ],
      }),
    ).toThrowErrorMatchingSnapshot();
  });
});
