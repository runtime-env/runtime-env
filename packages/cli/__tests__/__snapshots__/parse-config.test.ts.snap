// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`genJs.*.mode invalid 1`] = `
"[
  {
    "code": "custom",
    "message": "[@runtime-env/cli] SyntaxError: No duplicate modes allowed",
    "path": [
      "genJs"
    ]
  }
]"
`;

exports[`globalVariableName invalid 1`] = `
"[
  {
    "code": "custom",
    "message": "[@runtime-env/cli] SyntaxError: Invalid variable name",
    "path": [
      "globalVariableName"
    ]
  }
]"
`;
